integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import '../../solana-library/spl_token.sol';
+import '../../solana-library/system_instruction.sol';
+
+contract SimpleCollectible {
+    // Contract variables
+    uint32 private tokenCounter;
+    address private mint_authority;
+    address private payer;
+
+    struct NFTData {
+        address owner;
+        address mint_account;
+        string uri;
+        bool exists;
+    }
+
+    event NFTMinted(address owner, address mint_account, uint32 nft_id);
+    event NFTSold(address from, address to, uint32 nft_id);
+
+    mapping(uint32 => NFTData) private nft_information;
+
+    constructor (address _mint_authority) {
+        tokenCounter = 0;
+        // For every mint on Solana, a mint authority is needed to sign it
+        mint_authority = _mint_authority;
+    }
+
+    // Sets the payer account that is going to sign for every ownership transfer
+    function set_payer(address _payer) public {
+        payer = _payer;
+    }
```

Comment:
Why is the same for everyone?

Reply:
The payer is the account in Typescript used to derive the associated token account for each owner. It is the signer for the derived account and for each ownership transfer. As the payer account contains funds the belongs to the owner of this contract, I thought it should be the one to sign all transactions too.